opal/mca/common/ofi/common_ofi.c

```diff
@@ -675,8 +680,31 @@ static bool is_near(pmix_device_distance_t *distances,
 
     return false;
 }
-#endif
-#endif  // OPAL_OFI_PCI_DATA_AVAILABLE
+#endif /* HWLOC_API_VERSION < 0x00020000 */
+
+static bool is_near(pmix_device_distance_t *distances,
+                    int num_distances,
+                    struct fi_info *fi_info)
+{
+    /* if we failed to find any distances, then we consider all interfaces
+     * to be of equal distances and let the caller decide how to handle
+     * them
+     */
+    if (!distances)
+        return true;
+
+    if (fi_info && fi_info->fabric_attr && fi_info->fabric_attr->prov_name
+        && 0 == strcasecmp("efa", fi_info->fabric_attr->prov_name)) {
```

Comment:
Ugh; I don't understand this comment or why we're using ibv information here; let's step back and assume we've gone wrong somewhere and try again.

Reply:
i guess I don't understand why we're using NodeGUID and SysImageGUID for that at all, or why they're 0.  BUt I object to there being EFA-specific code here, as there's nothing about EFA that should make it need special code here.

Reply:
@bwbarrett I have refactored the logic that's incompatible with EFA, such that we do not need to call out EFA(or other vendor) in particular.

@amirshehataornl I reckon this conflicted with your previous refactor. Could you take a look and test it out on your system? FWICT it also addresses your previous concern with cpuset intersection with the wrong L3 cache.